mlir-sycl/lib/Dialect/IR/SYCLOpsAlias.cpp

```diff
@@ -67,6 +67,10 @@ SYCLOpAsmInterface::getAlias(mlir::Type Type, llvm::raw_ostream &OS) const {
     OS << "sycl_group_" << Group.getDimension();
     return AliasResult::FinalAlias;
   }
+  if (const auto GetScalarOp = Type.dyn_cast<mlir::sycl::GetScalarOpType>()) {
+    OS << "sycl_get_scalar_op_" << GetScalarOp.getDataType();
```

Comment:
```suggestion
    OS << "sycl_get_scalar_op_" << GetScalarOp.getDataType() << "_";
```

We don't want to have numeric characters at the end of alias ([see](https://github.com/intel/llvm/pull/7559#discussion_r1034475004)). Type could be, e.g., `i32`.

Reply:
If you look at the test cases, actually the alias has `_` at the end.